networkx/tests/test_lazy_imports.py

```diff
@@ -0,0 +1,96 @@
+import sys
+import importlib
+import types
+import pytest
+
+import networkx.lazy_imports as lazy
+
+
+def test_lazy_load_basics():
+    math = lazy.load("math")
+    anything_not_real = lazy.load("anything_not_real")
+
+    # Now test that accessing attributes does what it should
+    assert math.sin(math.pi) == pytest.approx(0, 1e-6)
+    # poor-mans pytest.raises for testing errors on attribute access
+    try:
+        anything_not_real.pi
+        assert False  # Should not get here
+    except ModuleNotFoundError:
```

Comment:
I guess these are not raised any more

Reply:
I think the tests pass, so these ModuleNotFoundError errors are raised.
Are you saying that they shouldn't be raised? 
Once we try to access an attribute of a lazily loaded module the module must finish the loading process right?

Reply:
Ah, woops, I thought you had changed it to `DelayedImportError`, my bad.